src/azure-cli/azure/cli/command_modules/appconfig/_snapshot_custom_client.py

```diff
@@ -0,0 +1,540 @@
+    name_filter=None,
+    status_filter=None,
+    select=None,
+    sync_token=None,
+    **kwargs
+):
+    _params = {}
+
+    if name_filter is not None:
+        _params['name'] = name_filter
+
+    if status_filter is not None:
+        _params['status'] = status_filter
+
+    if select is not None:
+        _params["$Select"] = Serializer().query("select", select, "[str]", div=",")
+
+    return _build_request(
+        "/snapshots",
+        RequestMethod.GET,
+        sync_token=sync_token,
+        params=_params,
+        **kwargs
+    )
+
+
+def build_status_update_request(
+    name,
+    archive_snapshot,
+    if_match=None,
+    if_none_match=None,
+    sync_token=None,
+    **kwargs
+):
+    request_body = {"status": "archived" if archive_snapshot else "ready"}
+
+    return _build_request(
+        "/snapshots/{name}",
+        RequestMethod.PATCH,
+        path_arguments={"name": name},
+        json=request_body,
+        if_match=if_match,
+        if_none_match=if_none_match,
+        sync_token=sync_token,
+        **kwargs
+    )
+
+
+def build_put_snapshot_request(
+    name,
+    filters,
+    composition_type=None,
+    retention_period=None,
+    tags=None,
+    if_match=None,
+    if_none_match=None,
+    sync_token=None,
+    **kwargs
+):
+
+    request_body = {}
+
+    if not filters or len(filters) < 1:
+        raise ValueError("There should be at least one filter specified.")
+
+    request_body["filters"] = filters
+
+    if composition_type is not None:
+        if composition_type not in ("all", "group_by_key"):
+            raise ValueError("Value should either be 'all' or 'group_by_key'.")
+
+        request_body["composition_type"] = composition_type
+
+    if retention_period is not None:
+        if not isinstance(retention_period, int) or retention_period < 0:
+            raise ValueError("Retention period value should be a non-negative integer value.")
+
+        request_body["retention_period"] = retention_period
+
+    if tags:
+        request_body["tags"] = tags
+
+    return _build_request(
+        "/snapshots/{name}",
+        RequestMethod.PUT,
+        path_arguments={"name": name},
+        json=request_body,
+        if_match=if_match,
+        if_none_match=if_none_match,
+        sync_token=sync_token,
+        **kwargs
+    )
+
+
+def build_list_snapshot_keys_request(
+    name,
+    select=None,
+    sync_token=None,
+    **kwargs
+):
+    _params = {"snapshot": name}
+
+    if select is not None:
+        _params["$Select"] = Serializer().query("select", select, "[str]", div=",")
+
+    return _build_request(
+        "/kv",
+        RequestMethod.GET,
+        sync_token=sync_token,
+        params=_params,
+        **kwargs
+    )
+
+
+def _convert_request(request, files=None):
+    data = request.content if not files else None
+    request = HttpRequest(method=request.method, url=request.url, headers=request.headers, data=data)
+    if files:
+        request.set_formdata_body(files)
```

Comment:
We don't use form data for any of our endpoints.

Reply:
Got it! This has been updated.